src/containers/onboarding/index.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from 'react';
+import { Button, Card, Form, Input, Typography } from 'antd';
+import styled from 'styled-components';
+import { ResponseCard } from '../../components/onboarding';
+
+const { Title, Paragraph } = Typography;
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  align-items: center;
+  width: 100%;
+`;
+
+const FormCard = styled(Card)`
+  display: flex;
+  width: 50%;
+  border-radius: 5px;
+`;
+
+const OnboardingPageTitle = styled(Title)`
+  display: flex;
+  width: 100%;
+  justify-content: center;
+`;
+
+const FormInput = styled(Input)`
+  display: flex;
+  width: 80%;
+  border: 1px 1px black;
+`;
+
+const StyledButton = styled(Button)`
+  width: 30%;
+  margin-top: 16px;
+`;
+
+const ResponseContainer = styled.div`
+  width: 75%;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  justify-content: space-evenly;
+  margin-top: 16px;
+`;
+
+const SuccessMessage = styled(Paragraph)`
+  color: green;
+`;
+
+interface OnboardingRequestData {
+  favoriteColor: string;
+}
+
+interface OnboardingResponseData {
+  id: number;
+  title: string;
+  body: string;
+  userId: number;
+}
+
+const Onboarding: React.FC = () => {
+  const [posts, setPosts] = useState<OnboardingResponseData[]>([]);
+  const [formSuccess, setFormSuccess] = useState<boolean>(false);
+
+  const updatePosts = async () => {
+    await fetch('https://jsonplaceholder.typicode.com/posts', {
```

Comment:
We should be showing the AsyncRequest/useState pattern that became idiomatic in the code, something like...

```
const [posts, setPosts] = useState<AsyncRequest<Post[], any>>(AsyncRequestNotStarted())

useEffect(()=> {

try {
setPosts(AsyncRequestLoading());
const result = setPosts(await apiClient.getPosts()
setPosts(AsyncRequestCompleted(posts))
} catch (err) {
setPosts(AsyncRequestFailed(err))
}
}

Reply:
well technically you cant use await in useEffect but either wrap it in an async function and call it or promise chain instead with .then and .catch